core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/CompressionAsyncRequestBody.java

```diff
@@ -140,73 +150,16 @@ public Builder chunkSize(Integer chunkSize) {
 
     private static final class SplittingSubscriber extends DelegatingSubscriber<ByteBuffer, Iterable<ByteBuffer>> {
         private final ChunkBuffer chunkBuffer;
-        private final AtomicBoolean upstreamDone = new AtomicBoolean(false);
-        private final AtomicLong downstreamDemand = new AtomicLong();
-        private final Object lock = new Object();
-        private volatile boolean sentFinalChunk = false;
 
-        protected SplittingSubscriber(Subscriber<? super Iterable<ByteBuffer>> subscriber, int chunkSize) {
+        protected SplittingSubscriber(Subscriber<? super Iterable<ByteBuffer>> subscriber, ChunkBuffer chunkBuffer) {
```

Comment:
Do we need `chunkBuffer` in this class?

Reply:
Right not needed, removed from SplittingSubscriber